### ISSUE TITLE
Cloud foundry deployment supports multiple buildpacks

### ISSUE THREAD

**Issue by sabbyanandan** [spring-cloud/spring-cloud-deployer-cloudfoundry#327]
@thienbaole commented on [Fri Nov 15 2019](https://github.com/spring-cloud/spring-cloud-dataflow/issues/3637)

Can we also support multiple buildpacks for task/stream deployment?
https://docs.cloudfoundry.org/buildpacks/use-multiple-buildpacks.html

**Comment by sabbyanandan** [comment]
Hi, @thienbaole. SCDF and the foundation are Java-based. There are a few other options to deploy and orchestrate polyglot workloads in SCDF, however (see [options](https://dataflow.spring.io/docs/recipes/polyglot/)). 

That said, what specifically are your use-cases? Would any of the above-described options work for these use-cases? If not, it'd be great if you could share the reasoning.

**Comment by thienbaole** [comment]
We need to deploy the same application with multiple buildpacks:
- First buildpack is customized to build the container
- Second buildpack is the standard one, java_buildpack

In the manifest we can specify 2 buildpacks as below:
```
applications:
- name: 
  memory: 
  path: 
  buildpacks:
    - customized_buildpack
    - java_buildpack
...
```

**Comment by sabbyanandan** [comment]
Thanks. I can see how this would be useful for some users. We will consider it for the next minor release of SCDF.

**Comment by sabbyanandan** [comment]
Relates to https://github.com/spring-cloud/spring-cloud-skipper/issues/938

**Comment by ilayaperumalg** [comment]
Looks like the CF Java client doesn't have the support for adding `buildpacks` attribute to the application manifest yet. Created an issue here: https://github.com/cloudfoundry/cf-java-client/issues/1037

**Comment by jvalkeal** [comment]
Reopening as things kinda went south with cf-java-client and netty. Changes in a cf-java-client to support multiple buildpacks got burned by an issue in a netty `4.1.51.Final` which is given from boot `2.2.9` and `2.3.3`.

Dataflow `2.6.0` was on boot `2.2.7` thus giving netty `4.1.49.Final`. Backporting this fix to `2.6.x` would then not work as we'd upgrade boot as well.

Fix has been merged to netty per https://github.com/netty/netty/pull/10481 which lands to next version `4.1.52.Final`. 

Error we're currently seeing with dataflow `2.7.0-SNAPSHOT` is:
```
 2020-08-11T18:19:13.22+0100 [APP/PROC/WEB/0] OUT reactor.netty.http.client.PrematureCloseException: Connection prematurely closed BEFORE response
   2020-08-11T18:19:13.22+0100 [APP/PROC/WEB/0] OUT 	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
   2020-08-11T18:19:13.22+0100 [APP/PROC/WEB/0] OUT Assembly trace from producer [reactor.core.publisher.MonoDoFinallyFuseable] :
   2020-08-11T18:19:13.22+0100 [APP/PROC/WEB/0] OUT 	reactor.core.publisher.Mono.checkpoint(Mono.java:1879)
   2020-08-11T18:19:13.22+0100 [APP/PROC/WEB/0] OUT 	org.cloudfoundry.reactor.client.v2.applications.ReactorApplicationsV2.upload(ReactorApplicationsV2.java:245)
   2020-08-11T18:19:13.22+0100 [APP/PROC/WEB/0] OUT Error has been observed at the following site(s):
   2020-08-11T18:19:13.22+0100 [APP/PROC/WEB/0] OUT 	|_ Mono.checkpoint ⇢ at org.cloudfoundry.reactor.client.v2.applications.ReactorApplicationsV2.upload(ReactorApplicationsV2.java:245)
   2020-08-11T18:19:13.22+0100 [APP/PROC/WEB/0] OUT 	|_ Mono.checkpoint ⇢ at org.cloudfoundry.operations.applications.DefaultApplications.pushManifest(DefaultApplications.java:431)
```

**Comment by jvalkeal** [comment]
Closing as things work now.
